backend/src/services/ChatBotPipeline.py

```diff
@@ -0,0 +1,112 @@
+from typing import Union

```

Comment:
Could you add a simple string explanation of what each method does as in num_tokens_from_string?